src/utils/utils.js

```diff
@@ -0,0 +1,25 @@
+import { redirect } from "react-router-dom";
+
+function getLoggedInUser() {
+  return JSON.parse(localStorage.getItem("user"));
+}
+function getToken() {
+  return JSON.parse(localStorage.getItem("user"))?.token;
+}
+
+function checkSession() {
+  return getToken() ? true : false;
```

Comment:
you should check getToken's value here, instead of just checking if value exist or not. if localstorage has value 1, it would be considered as true.